main.go

```diff
@@ -0,0 +1,231 @@
+		Definition: schema,
+	}
+
+	// Unmarshal message value and assign it to schemaStruct
+	fmt.Printf("Unmarshalling message for offset: %v \n", msg.Offset)
+	if err = consumerAvro.Unmarshal(msg.Value, schemaStruct); err != nil {
+		fmt.Errorf("error unmarshalling avro message for offset: %v, %v \n", msg.Offset, err)
+		os.Exit(1)
+	}
+	fmt.Printf("Message successfully unmarshalled for offset: %v \n", msg.Offset)
+
+	// Print the unmarshalled message out to the terminal
+	fmt.Println("Unmarshalled message:")
+	data, err := json.Marshal(schemaStruct)
+	if err != nil {
+		fmt.Errorf("error marshalling JSON message for offset: %v, %v \n", msg.Offset, err)
+		os.Exit(1)
+	}
+	fmt.Println(string(data))
+
+	// create avro producer
+	producerAvro := &avro.Schema{
+		Definition: schema,
+	}
+
+	// Marshall schemaStruct that contains message value ready for republishing
+	fmt.Printf("Marshalling message for offset: %v \n", msg.Offset)
+	message, err := producerAvro.Marshal(schemaStruct)
+	if err != nil {
+		fmt.Errorf("error marshalling avro message for offset: %v, %v \n", msg.Offset, err)
+		os.Exit(1)
+	}
+	fmt.Printf("Message successfully marshalled for offset: %v \n", msg.Offset)
+
+	// output the marshalled message into the messageBytes chan
+	messageBytes <- message
+}
+
+// Validates the flags to make sure all mandatory flags have been supplied
+// throw an error if not the case
+func validateFlags() {
+	flag.VisitAll(func(f *flag.Flag) {
+		if string(f.Value.String()) == "" {
+			fmt.Printf("Value not supplied for: %v \n", f.Name)
+			os.Exit(1)
+		}
+	})
+}
+
+// Prints the flags and parameters of the tool
+func printFlags() {
+	fmt.Println("---------------------")
+	fmt.Println("Parameters:")
+	flag.VisitAll(func(f *flag.Flag) {
+		fmt.Printf("%v: %v\n", f.Name, f.Value)
+	})
+	fmt.Println("---------------------")
+}
+
+// The offset slice is passed into method and is iterated over and each message
+// related to the offset in the topic/partition is outputted into the 'out' chan
+func consumePartition(consumer sarama.Consumer, topic string, partition int32, out chan *sarama.ConsumerMessage, offsetArray []int64) {
+	for _, offset := range offsetArray {
+		partitionConsumer, err := consumer.ConsumePartition(topic, partition, int64(offset))
+		if err != nil {
+			panic(err)
+		}
+
+		select {
+		case msg := <-partitionConsumer.Messages():
+			partitionConsumer.Close()
+			out <- msg
+		}
+	}
+}
+
+// This creates the offset array depending on whether a range or a single value
+// is entered as an argument to the tool
+func createOffsetArray(offset string) []int64 {
+	arraySize := make([]int64, 0)
+	if strings.ContainsAny(offset, "-") {
+		slice := strings.Split(offset, "-")
+		minRange, err := strconv.ParseInt(slice[0], 10, 64)
+		maxRange, err := strconv.ParseInt(slice[1], 10, 64)
+		if err != nil {
+			panic(err)
+		}
+
+		index := 0
+		for value := minRange; value <= maxRange; value++ {
+			arraySize = append(arraySize, value)
+			index++
```

Comment:
`index` not used - delete

Reply:
Resolved by [commit](https://github.com/companieshouse/kafka-management-tool/pull/1/commits/6cb5a49fcd24600087b3128739ed504aae6265d3)